aioamqp/compat.py

```diff
@@ -9,3 +9,17 @@
     from asyncio import ensure_future
 except ImportError:
     ensure_future = asyncio.async
+
+
+def iscoroutinepartial(fn):
+    # http://bugs.python.org/issue23519
+
+    while True:
+        parent = fn
+
+        fn = getattr(parent, 'func', None)
```

Comment:
while this code seems just fine, it would be better to update the state in one line like this:
```python
parent, fn = fn, getattr(parent, 'func', None)
```
see https://youtu.be/OSGv2VnC0go?t=34m for lengthier explanation